### PR TITLE
Enable non-valued variable defines

### LINKED ISSUE #128: Support non-valued BLD defines, e.g. "-D E1000_ENABLE" without =<foo>
The EDK2 [OvmfPkg README](https://github.com/tianocore/edk2/blob/master/OvmfPkg/README) instructs users to add non-valued "-D" defines to enable features.  

OvmfPkg/README:
>  - Add "-D E1000_ENABLE" to your build command (only when building
>      "OvmfPkg/OvmfPkgIa32X64.dsc" or "OvmfPkg/OvmfPkgX64.dsc").
>    - For example: "build -D E1000_ENABLE".

stuart_build throws an exception if I do not pass a `=<value>` in a BLD_ parameter, for example:

```
stuart_build -c PlatformBuild.py BLD_*_E1000_ENABLE

Traceback (most recent call last):
  File "C:\Program Files\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Program Files\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\git\venv_pytools\Scripts\stuart_build.exe\__main__.py", line 7, in <module>
  File "c:\git\venv_pytools\lib\site-packages\edk2toolext\invocables\edk2_platform_build.py", line 141, in main
    Edk2PlatformBuild().Invoke()
  File "c:\git\venv_pytools\lib\site-packages\edk2toolext\base_abstract_invocable.py", line 107, in Invoke
    self.ParseCommandLineOptions()
  File "c:\git\venv_pytools\lib\site-packages\edk2toolext\edk2_invocable.py", line 203, in ParseCommandLineOptions
    raise RuntimeError(f"Unknown variable passed in via CLI: {argument}")
RuntimeError: Unknown variable passed in via CLI: BLD_*_E1000_ENABLE
```

A workaround is to specify a dummy value **`=1`**, for example:
`stuart_build -c PlatformBuild.py BLD_*_E1000_ENABLE=1`

### PR DESCRIPTION
**Pull request by Javagedes**
Enable providing non-valued build defines and variable defines from the command line for stuart commands.

Resolves #128 

Signed-off-by: Joey Vagedes joeyvagedes@microsoft.com

### CODE REVIEW

**Comment by makubacki** [comment]
Changes look good, but are you going to add a unit test?

File: edk2toolext/edk2_invocable.py [hunk 1]
```diff
@@ -333,10 +333,13 @@ def ParseCommandLineOptions(self):
         epilog = dedent('''\
             positional arguments:
               <key>=<value>              - Set an env variable for the pre/post build process
+              <key>                      - Set a non-valued env variable for the pre/post build process
               BLD_*_<key>=<value>        - Set a build flag for all build types
                                            (key=value will get passed to build process)
+              BLD_*_<key>                - Set a non-valued build flag for all build types to TRUE
```
Comment (Javagedes) [inline edk2toolext/edk2_invocable.py:339]: remove "to TRUE". does not matter.

**Comment by Javagedes** [comment]
> Changes look good, but are you going to add a unit test?

Ah yes, I will add unit tests for this change.

File: edk2toolext/edk2_invocable.py [hunk 2]
```diff
@@ -422,10 +425,14 @@ def ParseCommandLineOptions(self):
...
-            if argument.count("=") != 1:
-                raise RuntimeError(f"Unknown variable passed in via CLI: {argument}")
-            tokens = argument.strip().split("=")
-            env.SetValue(tokens[0].strip().upper(), tokens[1].strip(), "From CmdLine")
+            if argument.count("=") == 1:
+                tokens = argument.strip().split("=")
+                env.SetValue(tokens[0].strip().upper(), tokens[1].strip(), "From CmdLine")
+                continue
+            elif argument.count("=") == 0:
+                env.SetValue(argument.strip().upper(), "TRUE", "From CmdLine")
```
Comment (spbrogan) [inline edk2toolext/edk2_invocable.py:433]: as we discussed this should be changed to either something totally random or some sort of string that indicates this value can not be used for value passing and should only be used for ifdef/ifndef like directives.

File: edk2toolext/tests/test_edk2_setup.py [hunk 3]
```diff
@@ -79,3 +79,41 @@ def test_setup_bad_omnicache_path(self):
...
+        try:
+            builder.Invoke()
+        except SystemExit as e:
+            self.assertEqual(e.code, 0)
+
+        env = shell_environment.GetBuildVars()
+        self.assertEqual(env.GetValue("BLD_*_VAR"), "TRUE")
+        self.assertEqual(env.GetValue("VAR"), "TRUE")
+        self.assertEqual(env.GetValue("BLD_DEBUG_VAR2"), "TRUE")
+        self.assertEqual(env.GetValue("BLD_RELEASE_VAR2"), "TRUE")
```
Comment (spbrogan) [inline edk2toolext/tests/test_edk2_setup.py:104]: these 4 should only be testing to make sure they are defined.  Not their value.

File: edk2toolext/edk2_invocable.py (hunk 2 above)
Comment (Javagedes) [reply edk2toolext/edk2_invocable.py]: @spbrogan I have updated the value of the variable to be a random string, and also updates the debug comment to state "Non Valued Variable Set From CmdLine"

I'm assuming ifdef/ifndef, you mean something along the lines of:

```python

if env.GetValue("NON_VALUED_VAR"):
  print("It exists")

if not env.GetValue("NON_VALUED_VAR"):
  print("It does not exist")
```

Then yes, this meets your expectations. Let me know!

File: edk2toolext/tests/test_edk2_setup.py (hunk 3 above)
Comment (Javagedes) [reply edk2toolext/tests/test_edk2_setup.py]: Updated to check verify that it exists via `self.assertIsNotNone()`

**Review by spbrogan — APPROVED** [review]
please add a few comments so a casual reader understands why it is a random 20 letter value

**Pull request merged**